dotnet/CoreLib/DataFormats/Image/AzureFormRecognizer/AzureFormRecognizerEngine.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Azure;
+using Azure.AI.FormRecognizer.DocumentAnalysis;
+
+namespace Microsoft.SemanticMemory.DataFormats.Image.AzureFormRecognizer;
+
+/// <summary>
+/// OCR engine based on Azure.AI.FormRecognizer.
+/// </summary>
+public class AzureFormRecognizerEngine : IOcrEngine
+{
+    private readonly DocumentAnalysisClient recognizerClient;
+
+    /// <summary>
+    /// Creates a new instance of the AzureFormRecognizerOcrEngine passing in the Form Recognizer endpoint and key.
+    /// </summary>
+    /// <param name="endpoint">The endpoint for accessing a provisioned Azure Form Recognizer instance</param>
+    /// <param name="credential">The AzureKeyCredential containing the provisioned Azure Form Recognizer access key</param>
+    public AzureFormRecognizerEngine(string endpoint, AzureKeyCredential credential)
```

Comment:
if we pass in `AzureFormRecognizerConfig` it will be easier to deal with different auth types, without the need to define multiple ctors, e.g. see `AzureTextGeneration`